sources/rds/db_instance.go

```diff
@@ -109,266 +108,407 @@ func dBInstanceOutputMapper(scope string, _ *rds.DescribeDBInstancesInput, outpu
+						Query:  *instance.MonitoringRoleArn,
+						Scope:  sources.FormatScope(a.AccountID, a.Region),
+					},
+					BlastPropagation: &sdp.BlastPropagation{
+						// Changing the role can affect the instance
+						In: true,
+						// The instance won't affect the role
+						Out: false,
+					},
+				})
 			}
 		}
 
 		if instance.PerformanceInsightsKMSKeyId != nil {
 			// This is an ARN
 			if a, err = sources.ParseARN(*instance.PerformanceInsightsKMSKeyId); err == nil {
 				// +overmind:link kms-key
-				item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{Query: &sdp.Query{
-					Type:   "kms-key",
-					Method: sdp.QueryMethod_SEARCH,
-					Query:  *instance.PerformanceInsightsKMSKeyId,
-					Scope:  sources.FormatScope(a.AccountID, a.Region),
-				}})
+				item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+					Query: &sdp.Query{
+						Type:   "kms-key",
+						Method: sdp.QueryMethod_SEARCH,
+						Query:  *instance.PerformanceInsightsKMSKeyId,
+						Scope:  sources.FormatScope(a.AccountID, a.Region),
+					},
+					BlastPropagation: &sdp.BlastPropagation{
+						// Changing the KMS key can affect the instance
+						In: true,
+						// The instance won't affect the KMS key
+						Out: false,
+					},
+				})
 			}
 		}
 
 		for _, role := range instance.AssociatedRoles {
 			if role.RoleArn != nil {
 				if a, err = sources.ParseARN(*role.RoleArn); err == nil {
 					// +overmind:link iam-role
-					item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{Query: &sdp.Query{
-						Type:   "iam-role",
-						Method: sdp.QueryMethod_SEARCH,
-						Query:  *role.RoleArn,
-						Scope:  sources.FormatScope(a.AccountID, a.Region),
-					}})
+					item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+						Query: &sdp.Query{
+							Type:   "iam-role",
+							Method: sdp.QueryMethod_SEARCH,
+							Query:  *role.RoleArn,
+							Scope:  sources.FormatScope(a.AccountID, a.Region),
+						},
+						BlastPropagation: &sdp.BlastPropagation{
+							// Changing the role can affect the instance
+							In: true,
+							// The instance won't affect the role
+							Out: false,
+						},
+					})
 				}
 			}
 		}
 
 		if instance.ActivityStreamKinesisStreamName != nil {
 			// +overmind:link kinesis-stream
-			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{Query: &sdp.Query{
-				Type:   "kinesis-stream",
-				Method: sdp.QueryMethod_GET,
-				Query:  *instance.ActivityStreamKinesisStreamName,
-				Scope:  scope,
-			}})
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "kinesis-stream",
+					Method: sdp.QueryMethod_GET,
+					Query:  *instance.ActivityStreamKinesisStreamName,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Tightly coupled
+					In:  true,
+					Out: true,
+				},
```

Comment:
How would the monitoring log stream affect the db instance?

Reply:
Sam kind of logic as `log-strem`